src/Ocelot/Configuration/Creator/SecurityOptionsCreator.cs

```diff
@@ -5,35 +5,26 @@ namespace Ocelot.Configuration.Creator
 {
     public class SecurityOptionsCreator : ISecurityOptionsCreator
     {
-        public SecurityOptions Create(FileSecurityOptions securityOptions)
-        {
-            var ipAllowedList = new List<string>();
-            var ipBlockedList = new List<string>();
+        public SecurityOptions Create(FileSecurityOptions securityOptions, FileGlobalConfiguration globalConfiguration) 
+            => Create(securityOptions.IsFullFilled() ? securityOptions : globalConfiguration.SecurityOptions);
 
-            foreach (var allowed in securityOptions.IPAllowedList)
-            {
-                if (IPAddressRange.TryParse(allowed, out var allowedIpAddressRange))
-                {
-                    var allowedIps = allowedIpAddressRange.Select<IPAddress, string>(x => x.ToString());
-                    ipAllowedList.AddRange(allowedIps);
-                }
-            }
-
-            foreach (var blocked in securityOptions.IPBlockedList)
-            {
-                if (IPAddressRange.TryParse(blocked, out var blockedIpAddressRange))
-                {
-                    var blockedIps = blockedIpAddressRange.Select<IPAddress, string>(x => x.ToString());
-                    ipBlockedList.AddRange(blockedIps);
-                }
-            }
+        private static SecurityOptions Create(FileSecurityOptions securityOptions)
+        {
+            var ipAllowedList = SetIpAddressList(securityOptions.IPAllowedList);
+            var ipBlockedList = SetIpAddressList(securityOptions.IPBlockedList);
 
             if (securityOptions.ExcludeAllowedFromBlocked)
             {
-                ipBlockedList = ipBlockedList.Except(ipAllowedList).ToList();
+                ipBlockedList = ipBlockedList.Except(ipAllowedList).ToArray();
             }
 
             return new SecurityOptions(ipAllowedList, ipBlockedList);
         }
-    }
+
+        private static string[] SetIpAddressList(IList<string> ipValueList)
+            => ipValueList
+                .Where(ipValue => IPAddressRange.TryParse(ipValue, out _))
+                .SelectMany(ipValue => IPAddressRange.Parse(ipValue).Select<IPAddress, string>(ip => ip.ToString()))
```

Comment:
Double parsing issue❗

Reply:
Solution:
```csharp
        private static string[] SetIpAddressList(List<string> ipValues) => ipValues.SelectMany(Parse).ToArray();
        private static string[] Parse(string ipValue)
        {
            if (IPAddressRange.TryParse(ipValue, out var range))
            {
                return range.Select<IPAddress, string>(ip => ip.ToString()).ToArray();
            }

            return Array.Empty<string>();
        }
```

Reply:
Please address all the issues in code review within 1 hour please!
I would like to merge it today and do not wish to waste any more time.

Reply:
fixed